### ISSUE TITLE
Failure when building the zstd library

### ISSUE THREAD

**Issue by mkauf** [envoyproxy/envoy#20923]
*Description*:
Since #20434, a local build of envoy fails with these messages:
```
ERROR: /home/user/.cache/bazel/_bazel_user/1fa8a1a6d1658496721262561afb60b4/external/envoy/bazel/foreign_cc/BUILD:551:12: output 'external/envoy/bazel/foreign_cc/zstd/lib/libzstd.a' was not created
ERROR: /home/user/.cache/bazel/_bazel_user/1fa8a1a6d1658496721262561afb60b4/external/envoy/bazel/foreign_cc/BUILD:551:12: Foreign Cc - CMake: Building zstd failed: not all outputs were created or valid
Target //source/exe:envoy-static failed to build
```

*Repro steps*:
On Fedora 35, build the current HEAD of envoy with:
```
bazel build --config=clang //:envoy
```

*Analysis*:
The zstd library is installed in the "lib64" directory, but the expected directory name is "lib".
```
$ bazel build --config=clang --verbose_failures --subcommands --sandbox_debug --jobs 1 //:envoy
[...]
[8/9] Install the project...
-- Install configuration: "Bazel"
-- Installing: /home/user/.cache/bazel/_bazel_user/1fa8a1a6d1658496721262561afb60b4/sandbox/linux-sandbox/6810/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy/bazel/foreign_cc/zstd/lib64/cmake/zstd/zstdTargets.cmake
-- Installing: /home/user/.cache/bazel/_bazel_user/1fa8a1a6d1658496721262561afb60b4/sandbox/linux-sandbox/6810/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy/bazel/foreign_cc/zstd/lib64/cmake/zstd/zstdTargets-bazel.cmake
-- Installing: /home/user/.cache/bazel/_bazel_user/1fa8a1a6d1658496721262561afb60b4/sandbox/linux-sandbox/6810/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy/bazel/foreign_cc/zstd/lib64/cmake/zstd/zstdConfig.cmake
-- Installing: /home/user/.cache/bazel/_bazel_user/1fa8a1a6d1658496721262561afb60b4/sandbox/linux-sandbox/6810/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy/bazel/foreign_cc/zstd/lib64/cmake/zstd/zstdConfigVersion.cmake
-- Installing: /home/user/.cache/bazel/_bazel_user/1fa8a1a6d1658496721262561afb60b4/sandbox/linux-sandbox/6810/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy/bazel/foreign_cc/zstd/lib64/pkgconfig/libzstd.pc
-- Installing: /home/user/.cache/bazel/_bazel_user/1fa8a1a6d1658496721262561afb60b4/sandbox/linux-sandbox/6810/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy/bazel/foreign_cc/zstd/include/zstd.h
-- Installing: /home/user/.cache/bazel/_bazel_user/1fa8a1a6d1658496721262561afb60b4/sandbox/linux-sandbox/6810/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy/bazel/foreign_cc/zstd/include/zdict.h
-- Installing: /home/user/.cache/bazel/_bazel_user/1fa8a1a6d1658496721262561afb60b4/sandbox/linux-sandbox/6810/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy/bazel/foreign_cc/zstd/include/zstd_errors.h
-- Installing: /home/user/.cache/bazel/_bazel_user/1fa8a1a6d1658496721262561afb60b4/sandbox/linux-sandbox/6810/execroot/envoy/bazel-out/k8-fastbuild/bin/external/envoy/bazel/foreign_cc/zstd/lib64/libzstd.a
[...]
```

*Proposed bugfix*:
The build succeeds with this change:
```
diff --git a/bazel/foreign_cc/BUILD b/bazel/foreign_cc/BUILD
index 82a2029086..6c59c4266d 100644
--- a/bazel/foreign_cc/BUILD
+++ b/bazel/foreign_cc/BUILD
@@ -553,6 +553,7 @@ envoy_cmake(
     build_data = ["@com_github_facebook_zstd//:all"],
     cache_entries = {
         "CMAKE_BUILD_TYPE": "Release",
+        "CMAKE_INSTALL_LIBDIR": "lib",
         "ZSTD_BUILD_SHARED": "off",
         "ZSTD_BUILD_STATIC": "on",
     },
```

**Comment by daixiang0** [comment]
Looks wried, it downloads with github archive link, make no distinction about OS, we can see it works in CI.

**Comment by wbpcode** [comment]
May cc @wrowe

**Comment by keith** [comment]
Fixed by https://github.com/envoyproxy/envoy/commit/54883cf195073d0ba95c992a1196b263c7c748de ?

**Comment by mkauf** [comment]
Yes, fixed by 54883cf195073d0ba95c992a1196b263c7c748de / #20918, thanks!
